### ISSUE TITLE
cargo-swift 0.8.0 installation fails due to uniffi_bindgen version mismatch

### ISSUE THREAD

**Issue by tszymczyszyn-shopify** [antoniusnaumann/cargo-swift#64]
Since the release of `uniffi-rs@0.28.1` installation of `cargo-swift@0.8.0` fails. Looks like they introduced a [breaking change](https://github.com/mozilla/uniffi-rs/pull/2213/files#diff-a2c8329d265afcb1ea0c4ba6f2ab1bb5cf71ab6be23d2a8a8cdc437adb5ca671R41) by altering the signature of `uniffi_bindgen::library_mode::generate_bindings`. 

<details>
  <summary>Output of `cargo install cargo-swift`</summary>

```
❯ cargo install cargo-swift@0.8.0
    Updating crates.io index
  Installing cargo-swift v0.8.0
    Updating crates.io index
     Locking 102 packages to latest compatible versions
      Adding cargo_metadata v0.15.4 (latest: v0.18.1)
      Adding encode_unicode v0.3.6 (latest: v1.0.0)
      Adding siphasher v0.3.11 (latest: v1.0.1)
      Adding toml v0.5.11 (latest: v0.8.19)
      Adding windows-sys v0.52.0 (latest: v0.59.0)
   Compiling proc-macro2 v1.0.86
   Compiling unicode-ident v1.0.12
   Compiling serde v1.0.207
   Compiling autocfg v1.3.0
   Compiling version_check v0.9.5
   Compiling memchr v2.7.4
   Compiling thiserror v1.0.63
   Compiling semver v1.0.23
   Compiling camino v1.1.7
   Compiling unicase v2.7.0
   Compiling serde_json v1.0.124
   Compiling ryu v1.0.18
   Compiling anyhow v1.0.86
   Compiling itoa v1.0.11
   Compiling fs-err v2.11.0
   Compiling libm v0.2.8
   Compiling libc v0.2.155
   Compiling minimal-lexical v0.2.1
   Compiling quote v1.0.36
   Compiling syn v2.0.74
   Compiling mime_guess v2.0.5
   Compiling nom v7.1.3
   Compiling num-traits v0.2.19
   Compiling typenum v1.17.0
   Compiling mime v0.3.17
   Compiling utf8parse v0.2.2
   Compiling anstyle-parse v0.2.5
   Compiling askama_parser v0.2.1
   Compiling bytes v1.7.1
   Compiling siphasher v0.3.11
   Compiling paste v1.0.15
   Compiling anstyle-query v1.1.1
   Compiling once_cell v1.19.0
   Compiling is_terminal_polyfill v1.70.1
   Compiling colorchoice v1.0.2
   Compiling portable-atomic v1.7.0
   Compiling unicode-width v0.1.13
   Compiling anstyle v1.0.8
   Compiling lazy_static v1.5.0
   Compiling execute-command-tokens v0.1.7
   Compiling smawk v0.3.2
   Compiling console v0.15.8
   Compiling textwrap v0.16.1
   Compiling weedle2 v5.0.0
   Compiling anstream v0.6.15
   Compiling humansize v2.1.3
   Compiling heck v0.5.0
   Compiling strsim v0.11.1
   Compiling log v0.4.22
   Compiling plain v0.2.3
   Compiling serde_derive v1.0.207
   Compiling thiserror-impl v1.0.63
   Compiling uniffi_checksum_derive v0.28.1
   Compiling scroll_derive v0.12.0
   Compiling execute-command-macro-impl v0.1.10
   Compiling clap_lex v0.7.2
   Compiling percent-encoding v2.3.1
   Compiling askama_escape v0.10.3
   Compiling clap_builder v4.5.15
   Compiling uniffi_meta v0.28.1
   Compiling execute-command-macro v0.1.9
   Compiling generic-array v1.1.0
   Compiling scroll v0.12.0
   Compiling clap_derive v4.5.13
   Compiling goblin v0.8.2
   Compiling glob v0.3.1
   Compiling unicode-segmentation v1.11.0
   Compiling number_prefix v0.4.0
   Compiling either v1.13.0
   Compiling shell-words v1.1.0
   Compiling execute v0.2.13
   Compiling clap v4.5.15
   Compiling dialoguer v0.11.0
   Compiling convert_case v0.6.0
   Compiling itertools v0.13.0
   Compiling indicatif v0.17.8
   Compiling nonempty v0.10.0
   Compiling basic-toml v0.1.9
   Compiling cargo-platform v0.1.8
   Compiling askama_derive v0.12.5
   Compiling toml v0.5.11
   Compiling cargo_metadata v0.15.4
   Compiling uniffi_testing v0.28.1
   Compiling uniffi_udl v0.28.1
   Compiling askama v0.12.1
   Compiling uniffi_bindgen v0.28.1
   Compiling cargo_metadata v0.18.1
   Compiling cargo-swift v0.8.0
error[E0061]: this function takes 7 arguments but 6 arguments were supplied
  --> /Users/tomasz/.cargo/registry/src/index.crates.io-6f17d22bba15001f/cargo-swift-0.8.0/src/bindings.rs:22:26
   |
22 |     let uniffi_outputs = uniffi_bindgen::library_mode::generate_bindings(
   |                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
...
26 |         None,
   |         ---- an argument of type `&dyn BindgenCrateConfigSupplier` is missing
   |
note: function defined here
  --> /Users/tomasz/.cargo/registry/src/index.crates.io-6f17d22bba15001f/uniffi_bindgen-0.28.1/src/library_mode.rs:37:8
   |
37 | pub fn generate_bindings<T: BindingGenerator + ?Sized>(
   |        ^^^^^^^^^^^^^^^^^
help: provide the argument
   |
22 |     let uniffi_outputs = uniffi_bindgen::library_mode::generate_bindings(lib_path, None, &SwiftBindingGenerator {}, /* &dyn BindgenCrateConfigSupplier */, None, out_dir, false)?;
   |                                                                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

For more information about this error, try `rustc --explain E0061`.
error: could not compile `cargo-swift` (lib) due to 1 previous error
warning: build failed, waiting for other jobs to finish...
error: failed to compile `cargo-swift v0.8.0`, intermediate artifacts can be found at `/var/folders/3w/z81qm8ns52z7w_cwbgf6yr7r0000gn/T/cargo-install2kO8op`.
To reuse those artifacts with a future compilation, set the environment variable `CARGO_TARGET_DIR` to that path.
```
</details>

The workaround for me was to use locally patched version of `cargo-swift` with `uniffi_bindgen = { version = "= 0.28.0" }` (instead of `version = "0.28.0"`).

**Comment by tszymczyszyn-shopify** [comment]
Another way to deal with this is to produce `Cargo.lock` (with `uniffi_bindgen` fixed at v0.28.0) and run `cargo install --locked cargo-swift@0.8.0`.

**Comment by antoniusnaumann** [comment]
Thanks for bringing this up. I'll publish a hotfix later this week

**Comment by antoniusnaumann** [comment]
This should be fixed in #65. I published the changes from this branch as v0.8.1. let me know if you encounter any issues

**Comment by tszymczyszyn-shopify** [comment]
Thanks @antoniusnaumann! Much appreciated :)
